cis194/week2/tien/homework2/src/LogAnalysis.hs

```diff
@@ -0,0 +1,47 @@
+{-# OPTIONS_GHC -Wall #-}
+
+module LogAnalysis where
+
+import           Log
+import           Text.Read (readMaybe)
+
+getHead :: [String] -> String
+getHead wordList =
+  case wordList of
+    x:_ -> x
+    _   -> ""
```

Comment:
When we find ourselves coming up with default value (here `""`) for a pattern match and that default value is not necessary for the requirement, we normally have the wrong design of our code. Please see comments on `parseMessage`.